dingo.go

```diff
@@ -693,6 +693,10 @@ func (injector *Injector) requestInjection(object interface{}, circularTrace []c
 
 		i++
 
+		if ctype.Kind() != reflect.Ptr && current.MethodByName("Inject").IsValid() {
+			return fmt.Errorf("usage of 'Inject' method with struct receiver is not allowed: %s", current.String())
```

Comment:
Can you define a global error at the beginning of dingo.go:
```go
var ErrInvalidInjectReceiver = errors.New("usage of 'Inject' method with struct receiver is not allowed")
```
then, change this line to
```go
return fmt.Errorf("invalid inject receiver %s: %w", current, ErrInvalidInjectReceiver)
```

Reply:
Done.